python-package/lightgbm/dask.py

```diff
@@ -282,17 +354,100 @@ def _train(
     data_parts = _split_to_parts(data=data, is_matrix=True)
     label_parts = _split_to_parts(data=label, is_matrix=False)
     parts = [{'data': x, 'label': y} for (x, y) in zip(data_parts, label_parts)]
+    n_parts = len(parts)
 
     if sample_weight is not None:
         weight_parts = _split_to_parts(data=sample_weight, is_matrix=False)
-        for i in range(len(parts)):
+        for i in range(n_parts):
```

Comment:
I totally support ths simplification. Could you make this a separate PR? Doesnt require new tests and we can merge it immediately

Reply:
Hey yes, you're just referring to the swapping of `len(parts)` for `n_parts`, right?

Reply:
yep exactly, just that little bit

Reply:
https://github.com/microsoft/LightGBM/pull/3985

Reply:
merged, thanks. Whenever you merge `master` back into this branch, those lines should drop out of the diff.